src/Microsoft.ServiceFabric.Actors/Runtime/MigrationOrchestrator.cs

```diff
@@ -0,0 +1,452 @@
+                return lastUpdatedRecord;
+            }
+
+            var worker = new MigrationWorker(this.stateProvider, this.actorTypeInfo);
+            cancellationToken.ThrowIfCancellationRequested();
+            while (endSequenceNumber - lastUpdatedRecord > this.downtimeThreshold)
+            {
+                catchupCount++;
+                cancellationToken.ThrowIfCancellationRequested();
+                using (var tx = this.stateProvider.GetStateManager().CreateTransaction())
+                {
+                    this.AddOrUpdateMetadata(tx, MigrationConstants.CatchupIterationKey, catchupCount.ToString());
+                    this.AddOrUpdateMetadata(tx, MigrationConstants.GetCatchupWorkerStartSNKey(catchupCount), lastUpdatedRecord + 1.ToString());
+                    this.AddOrUpdateMetadata(tx, MigrationConstants.GetCatchupWorkerEndSNKey(catchupCount), endSequenceNumber.ToString());
+                    await tx.CommitAsync();
+                }
+
+                await worker.StartMigrationWorker(MigrationPhase.Catchup, lastUpdatedRecord + 1, endSequenceNumber, catchupCount, cancellationToken);
+                lastUpdatedRecord = endSequenceNumber;
+                endSequenceNumber = long.Parse(await Client.GetStringAsync(this.kvsEndpoint + MigrationConstants.GetEndSNEndpoint));
+            }
+
+            return lastUpdatedRecord;
+        }
+
+        private async Task StartDowntimePhaseWorkload(long lastUpdatedRecord, CancellationToken cancellationToken)
+        {
+            await Client.PutAsync(this.kvsEndpoint + MigrationConstants.RejectWritesAPIEndpoint, null);
+
+            var endSequenceNumber = long.Parse(await Client.GetStringAsync(this.kvsEndpoint + MigrationConstants.GetEndSNEndpoint));
+            if (endSequenceNumber <= lastUpdatedRecord)
+            {
+                return;
+            }
+
+            var worker = new MigrationWorker(this.stateProvider, this.actorTypeInfo);
+
+            cancellationToken.ThrowIfCancellationRequested();
+            using (var tx = this.stateProvider.GetStateManager().CreateTransaction())
+            {
+                this.AddOrUpdateMetadata(tx, MigrationConstants.MigrationPhaseKey, MigrationPhase.Downtime.ToString());
+                this.AddOrUpdateMetadata(tx, MigrationConstants.DowntimeStartSNKey, lastUpdatedRecord + 1.ToString());
+                this.AddOrUpdateMetadata(tx, MigrationConstants.DowntimeEndSNKey, endSequenceNumber.ToString());
+                await tx.CommitAsync();
+            }
+
+            await worker.StartMigrationWorker(MigrationPhase.Downtime, lastUpdatedRecord + 1, endSequenceNumber, -1, cancellationToken);
+            await this.ResumeWritesAndMarkeMigrationCompleted(cancellationToken);
+        }
+
+        private async Task ResumeWritesAndMarkeMigrationCompleted(CancellationToken cancellationToken)
+        {
+            await Client.PutAsync(this.kvsEndpoint + MigrationConstants.ResumeWritesAPIEndpoint, null);
```

Comment:
You should resume writes on the KVS. ResumeWrites is a gesture that comes from the user if he/she is not happy with the migration.

Reply:
Updated to remove ResumeWrites